src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/HearingsJourneyService.java

```diff
@@ -0,0 +1,79 @@
+    private static final String UPDATED_SUMMARY = "SSCS - updated case sent to HMC";
+    private static final String UPDATED_DESC = "SSCS - updated case sent to HMC";
+    private static final String CANCELLED_SUMMARY = "SSCS - case cancelled";
+    private static final String CANCELLED_DESC = "SSCS - case cancelled";
+    private static final String EXCEPTION_SUMMARY = "SSCS - exception occurred";
+    private static final String EXCEPTION_DESC = "SSCS - exception occurred";
+
+    private final HmcHearingService hmcHearingService;
+    private final CcdCaseService ccdCaseService;
+    private final CcdStateUpdateService ccdStateUpdateService;
+    private final CcdLocationUpdateService locationUpdateService;
+
+    public void process(HmcMessage hmcMessage) throws GetCaseException, UpdateCaseException {
+        validateHmcMessage(hmcMessage);
+
+        final String hearingId = hmcMessage.getHearingID();
+        HearingGetResponse hearingResponse = hmcHearingService.getHearingRequest(hearingId);
+        if (isNull(hearingResponse)) {
+            throw new GetCaseException(String.format("Failed to retrieve hearing with Id: %s from HMC", hearingId));
+        }
+
+        SscsCaseData caseData = ccdCaseService.getCaseDetails(Long.parseLong(hearingId)).getData();
+
+        switch (hmcMessage.getHearingUpdate().getHmcStatus()) {
+            case LISTED:
+                ccdStateUpdateService.updateListed(hearingResponse, caseData);
+                locationUpdateService.updateVenue(hmcMessage, caseData);
+                ccdCaseService.updateCaseData(caseData, HEARING_BOOKED, CREATED_SUMMARY, CREATED_DESC);
+                break;
+            case UPDATE_SUBMITTED:
+                ccdStateUpdateService.updateListed(hearingResponse, caseData);
+                locationUpdateService.updateVenue(hmcMessage, caseData);
+                ccdCaseService.updateCaseData(caseData, UPDATE_CASE_ONLY, UPDATED_SUMMARY, UPDATED_DESC);
+                break;
+            case CANCELLED:
+                ccdStateUpdateService.updateCancelled(hearingResponse, caseData);
+                ccdCaseService.updateCaseData(caseData, UPDATE_CASE_ONLY, CANCELLED_SUMMARY, CANCELLED_DESC);
+                break;
+            case EXCEPTION:
+                ccdStateUpdateService.updateFailed(caseData);
+                ccdCaseService.updateCaseData(caseData, UPDATE_CASE_ONLY, EXCEPTION_SUMMARY, EXCEPTION_DESC);
+                break;
+            default:
+                log.info("CCD state has not been updated for the hearing Id: {}", hearingId);
+                break;
+        }
+    }
+
+    private void validateHmcMessage(HmcMessage hmcMessage) {
+        requireNonNull(hmcMessage, "HMC message must not be mull");
+        requireNonNull(hmcMessage.getHearingID(), "HMC message field hearingID is missing");
+        requireNonNull(hmcMessage.getHearingUpdate().getHmcStatus(), "HMC message field HmcStatus is missing");
+    }
```

Comment:
As mentioned above, I don't like this method of validating data.
Adding on to the above, typically NullPointerExceptions mean something has gone wrong, bad news, typically a coding error, so another reason not to use this technique